### ISSUE TITLE
User details are getting null 

### ISSUE THREAD

**Issue by grajuu** [jhipster/jdl-samples#24]
Hi @mraible ,

      I tried the following JDL, based on some of the examples.
-- I can able to run all the dockers and launch the application.
--I can able to Login with both User and Admin 
Issue is : 
  -- When am trying create a "Member "  from entity screen am getting an error saying Users cannot be null.

I did 

> One-to-One
relation ship with Members and Users

 
User details should pull from 

> Keycloak

 , if am not wrong.
Did i miss something ..?

Can you please help in this .
--Thanks.  

```
application {
  config {
    baseName gateway,
    packageName com.sample.gateway,
    applicationType gateway,
    authenticationType oauth2,
    prodDatabaseType postgresql,
    searchEngine elasticsearch,
    serviceDiscoveryType eureka,
    testFrameworks [protractor],
    buildTool maven,
    clientFramework react,
    nativeLanguage en,
    languages [en, es] 
  }
  entities Members, Rewards, Address, Product, ProductCategory, ProductOrder, OrderItem, Invoice, Shipment
}

application {
  config {
    baseName school,
    packageName com.sample.users,
    applicationType microservice,
    authenticationType oauth2,
    prodDatabaseType postgresql,
    searchEngine elasticsearch,
    buildTool maven,
    serverPort 8081,
    serviceDiscoveryType eureka
  }
  entities Members, Rewards, Address
}

application {
  config {
    baseName store,
    packageName com.sample.store,
    applicationType microservice,
    authenticationType oauth2,
    databaseType mongodb,
    devDatabaseType mongodb,
    prodDatabaseType mongodb,
    searchEngine elasticsearch,
    buildTool maven,
    enableHibernateCache false,
    serverPort 8082,
    serviceDiscoveryType eureka
  }
 entities Product, ProductCategory, ProductOrder, OrderItem
}

application {
  config {
    baseName invoice,
    applicationType microservice,
    packageName com.sample.invoice,
    authenticationType oauth2,
    prodDatabaseType postgresql,
    buildTool maven,
    serverPort 8083,
    serviceDiscoveryType eureka
  }
  entities Invoice, Shipment
}


// Customer for the store
entity Members {
  firstName String required,
    lastName String required,
    gender Gender required,
    who Who required,
    email String required pattern(/^[^@\s]+@[^@\s]+\.[^@\s]+$/)
    age BigDecimal
}
// Address table for both Student and Faculty 
entity Address{
  addressLine1 String,
    addressLine2 String,
    city String,
    country String,
    primaryPhone String ,
    secondryPhone String
}
enum Gender {
  MALE, FEMALE, OTHER
}
enum Who {
  STUDENT, FACULTY, ADMIN
}
entity Rewards {
  facultyName String ,
        studentName String,
        timestamp ZonedDateTime required,
        rewardReceved BigDecimal,
        date Instant required,
        content TextBlob required
}
/**
 * Entities for store microservice
 */
// Product sold by the Online store 
entity Product {
  name String required
    description String
    price BigDecimal required min(0)
    size Size required
    image ImageBlob
}
enum Size {
  S, M, L, XL, XXL
}
entity ProductCategory {
  name String required,
    description String
}
entity ProductOrder {
  placedDate Instant required,
    email String required pattern(/^[^@\s]+@[^@\s]+\.[^@\s]+$/),
    status OrderStatus required,
    code String required,
    invoiceId Long,
    customer String required
}
enum OrderStatus {
  COMPLETED, PENDING, CANCELLED
}
entity OrderItem {
  quantity Integer required min(0)
    totalPrice BigDecimal required min(0)
    status OrderItemStatus required
}
enum OrderItemStatus {
  AVAILABLE, OUT_OF_STOCK, BACK_ORDER
}
/* Entities for Invoice microservice */
entity Invoice {
  code String required
    date Instant required
    details String
    status InvoiceStatus required
    paymentMethod PaymentMethod required
    paymentDate Instant required
    paymentAmount BigDecimal required
}
enum InvoiceStatus {
  PAID, ISSUED, CANCELLED
}
entity Shipment {
  trackingCode String
    date Instant required
    details String
}
enum PaymentMethod {
  CREDIT_CARD, CASH_ON_DELIVERY, PAYPAL
}
/* Entities for notification microservice */
entity Notification {
  date Instant required
    details String
    sentDate Instant required
    format NotificationType required
    userId Long required
    productId Long required
}
enum NotificationType {
  EMAIL, SMS, PARCEL
}
relationship ManyToOne {
  OrderItem{product(name) required} to Product
}
relationship OneToMany {
  Members{address} to Address{members(email) required} ,
  Members{rewards} to Rewards{members(email) required} ,
  ProductOrder{orderItem} to OrderItem{order(code) required} , 
  ProductCategory{product} to Product{productCategory(name)},
  Invoice{shipment} to Shipment{invoice(code) required}
}
relationship OneToOne {
  Members{user(login) required} to User
}
service Invoice, Shipment, Product, ProductCategory, ProductOrder, OrderItem with serviceClass
paginate Invoice, Shipment, Product, Members, ProductOrder, OrderItem with pagination
// MicroServices 
microservice Product, ProductOrder, ProductCategory, OrderItem with store
microservice Members, Address, Rewards with school
microservice Invoice, Shipment with invoice
microservice Notification with notification
// will be created under 'docker-compose' folder
deployment {
  deploymentType docker-compose
  appsFolders [gateway, school, store, invoice]
  dockerRepositoryName "govindraju9999"
  consoleOptions [zipkin]
}

```

**Exception in Logger** 

`Bad Request: Validation failed for argument [0] in public org.springframework.http.ResponseEntity<com.samplel.users.domain.Members> com.sample.users.web.rest.MembersResource.createMembers(com.sample.users.domain.Members) throws java.net.URISyntaxException: [Field error in object 'members' on field 'user': rejected value [null]; codes [NotNull.members.user,NotNull.user,NotNull.com.sample.users.domain.User,NotNull]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [members.user,user]; arguments []; default message [user]]; default message [must not be null]] `

**Comment by mraible** [comment]
What version of JHipster are you using?

**Comment by grajuu** [comment]
```
$ jhipster --version
INFO! Using JHipster version installed globally
6.8.0

```

**Comment by mraible** [comment]
I'm curious: why did you include the output from `ignite doctor`? Are you creating a React Native app with Ignite JHipster, or are you just creating a React app with JHipster?

**Comment by grajuu** [comment]
You are right @mraible  ,
 Am creating react native as well  with Ignite JHipster,.

In react native Login redirection issue coming, then i realize there is separate place for that 👍  .
though am tag both issue to you :)

**Comment by mraible** [comment]
Do you see this issue in your React Native UI, or in JHipster? Tagging @ruddell in case it's React Native.

**Comment by grajuu** [comment]
Am facing this mapping issue in JHipster react web UI.

Sure i will tag @ruddell for Login redirection issue on react-native.

**Comment by grajuu** [comment]
Hey @mraible,  did i miss any configuration ..? or my JDL file have wrong mapping ..?

**Comment by mraible** [comment]
I'm not sure. I did fix an issue that seems similar recently. https://github.com/jhipster/generator-jhipster/issues/8920

**Comment by grajuu** [comment]
Hey @mraible 
So based on your comment 
https://github.com/jhipster/generator-jhipster/pull/11317

it works with Angular, but fails when you use React on the gateway..?

But it is on Feb 2020 ;)  Still this issue persists ..? :) @saturnism 

Can you guys please confirm.
--Thanks

**Comment by mraible** [comment]
Yes, the bug I fixed happened only in React and it was released with JHipster 6.8.0. If this issue is still happening, please [enter an issue in the main JHipster project](https://github.com/jhipster/generator-jhipster/issues/new?template=BUG_REPORT.md) and include your JDL.

I'm closing this issue here because the JDL you reference doesn't exist in this project.
